### PR TITLE
Move application details call to SWR

### PR DESCRIPTION
**Pull request by brionmario**
### Purpose

Visual Editor newly saved state is lost when navigating across tabs. This PR fixes the issue.

### Related Issues
- Address https://github.com/wso2/product-is/issues/17125

### Related PRs
- N/A

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/applications/pages/application-edit.tsx [hunk 1]
```diff
@@ -115,6 +116,43 @@ const ApplicationEditPage: FunctionComponent<ApplicationEditPageInterface> = (
...
+    const {
+        data: application,
+        mutate: mutateApplicationGetRequest,
+        isLoading: isApplicationGetRequestLoading,
+        error: applicationGetRequestError
+    } = useGetApplication(applicationId, !!applicationId);
+
+    useEffect(() => {
+        setApplicationRequestLoading(isApplicationGetRequestLoading);
+    }, [ isApplicationGetRequestLoading ]);
```
Comment (pavinduLakshan) [inline apps/console/src/features/applications/pages/application-edit.tsx:128]: Can't we directly use the loading variable returned by the swr hook?

File: apps/console/src/features/applications/pages/application-edit.tsx [hunk 2]
```diff
@@ -115,6 +116,43 @@ const ApplicationEditPage: FunctionComponent<ApplicationEditPageInterface> = (
...
+
+    useEffect(() => {
+        if (!applicationGetRequestError) {
+            return;
+        }
+
+        if (applicationGetRequestError.response
+            && applicationGetRequestError.response.data
+            && applicationGetRequestError.response.data.description) {
```
Comment (pavinduLakshan) [inline apps/console/src/features/applications/pages/application-edit.tsx:137]:
```suggestion
        if (applicationGetRequestError?.response?.data?.description) {
```

File: apps/console/src/features/applications/pages/application-edit.tsx [hunk 3]
```diff
@@ -338,50 +376,14 @@ const ApplicationEditPage: FunctionComponent<ApplicationEditPageInterface> = (
         if (application.templateId === ApplicationManagementConstants.CUSTOM_APPLICATION_OIDC
             || application.templateId === ApplicationManagementConstants.CUSTOM_APPLICATION_SAML
             || application.templateId === ApplicationManagementConstants.CUSTOM_APPLICATION_PASSIVE_STS) {
-            template = applicationTemplates.find((template: ApplicationTemplateListItemInterface) => 
+            template = applicationTemplates.find((template: ApplicationTemplateListItemInterface) =>
```
Comment (pavinduLakshan) [inline apps/console/src/features/applications/pages/application-edit.tsx:379]:
```suggestion
            template = applicationTemplates?.find((template: ApplicationTemplateListItemInterface) =>
```

File: apps/console/src/features/applications/pages/application-edit.tsx (hunk 1 above)
Comment (brionmario) [reply apps/console/src/features/applications/pages/application-edit.tsx:129]: There is a `setIsLoading` passed to the edit component and the nested form components to a very deep level and it is calling back to set the loading state of the parent. So, its not safe to refactor that at this time.

https://github.com/wso2/identity-apps/blob/master/apps/console/src/features/applications/components/edit-application.tsx#L658

File: apps/console/src/features/applications/pages/application-edit.tsx (hunk 2 above)
Comment (brionmario) [reply apps/console/src/features/applications/pages/application-edit.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/4283/commits/3165129d83031c70b28326b62ceb44b277e05373

File: apps/console/src/features/applications/pages/application-edit.tsx (hunk 3 above)
Comment (brionmario) [reply apps/console/src/features/applications/pages/application-edit.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/4283/commits/af042fbd2d2589041670f6c11d48ffee5ada9099

**Review by pavinduLakshan — APPROVED** [review]

**Pull request merged**